modules/nf-core/minimap2/align/main.nf

```diff
@@ -35,13 +36,17 @@ process MINIMAP2_ALIGN {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def bam_index = bam_index_extension ? "${prefix}.bam##idx##${prefix}.bam.${bam_index_extension} --write-index" : "${prefix}.bam"
     def map_mode = "${meta.platform}" ? "-x map-${meta.platform}" : ''
-    def bam_output = bam_format ? "-a | samtools fastq -f 4 | gzip > ${prefix}.${prefix2}.minimap.fastq.gz" : "-o ${prefix}.paf"
+    def bam_output = bam_format ? "-a | samtools ${fa_fq_format} -f 4 | gzip > ${prefix}.${prefix2}.minimap.${fa_fq_format}.gz" : "-o ${prefix}.paf"
+    def bam_real_output = fa_fq_format.matches('bam') ? "-a | samtools sort -@ ${task.cpus-1} -o ${bam_index} ${args2}" : ""
     def cigar_paf = cigar_paf_format && !bam_format ? "-c" : ''
     def set_cigar_bam = cigar_bam && bam_format ? "-L" : ''
     def bam_input = "${reads.extension}".matches('sam|bam|cram')
-    def samtools_reset_fastq = bam_input ? "samtools reset --threads ${task.cpus-1} $args3 $reads | samtools fastq --threads ${task.cpus-1} $args4 |" : ''
+    def samtools_reset_fastq = bam_input ? "samtools reset --threads ${task.cpus-1} $args3 $reads | samtools ${fa_fq_format} --threads ${task.cpus-1} $args4 |" : ''
     def query = bam_input ? "-" : reads
     def target = reference ?: (bam_input ? error("BAM input requires reference") : reads)
+    
+    if(bam_real_output != "")
```

Comment:
pretty sure you can replace with `if (!bam_real_output)` since an empty string asserts as false in nextflow

Reply:
Note to self: update meta.yml for minimap